src/docfx/lib/markdown/validation/ContentValidationExtension.cs

```diff
@@ -38,10 +42,39 @@ internal static class ContentValidationExtension
                             });
                         }
 
+                        if (node is InclusionBlock inclusionBlock)
+                        {
+                            var inclusionDocument = (Document?)readFile(inclusionBlock.IncludedFilePath, InclusionContext.File, inclusionBlock).file;
+                            if (inclusionDocument != null)
+                            {
+                                documentHeadings.Add(new Heading
+                                {
+                                    Level = -1,
+                                    SourceInfo = inclusionBlock.ToSourceInfo(),
+                                    Content = inclusionDocument.FilePath.ToString(),
+                                });
+                            }
+                        }
+
                         return false;
                     });
 
-                    contentValidator.ValidateHeadings((Document)InclusionContext.File, headings, InclusionContext.IsInclude);
+                    var allHeadings = getHeadings(documentHeadings);
+
+                    if (InclusionContext.IsInclude && documentHeadings.Any())
+                    {
+                        var rootFile = (Document)InclusionContext.RootFile;
+                        var rootFileHeadings = allHeadings.TryGetValue(rootFile, out var headings) ? headings.headings : null;
+                        if (rootFileHeadings != null)
+                        {
+                            var headingPlaceHolders = rootFileHeadings.Where(h => h.Level == -1 && h.Content == ((Document)InclusionContext.File).FilePath.ToString()).ToList();
+                            foreach (var headingPlaceHolder in headingPlaceHolders)
```

Comment:
how about `for` loop to manipulate `rootFileHeadings` directly?

Reply:
will use `while` instead of `for`